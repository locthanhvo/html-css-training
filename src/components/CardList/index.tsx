import { DragEvent, memo, useCallback, useMemo } from 'react';
import { Button, Flex, VStack } from '@chakra-ui/react';

// Components
import HeaderList from '@/components/common/HeaderList';
import CardItem from '@/components/CardItem';
import { AddTaskForm } from '@/components/Form';

// Types
import { IListTask, TAddTaskForm } from '@/types';

interface CardListProps {
  isLoading?: boolean;
  isActive?: boolean;
  taskList?: IListTask;
  onSubmit: (data: TAddTaskForm, imageFiles?: File[]) => void;
  onOpenAddTask?: (id: string) => void;
  onClick?: (id: string) => void;
  onTaskDrop: (taskId: string, targetListId: string) => void;
}

const CardList = ({
  isLoading = false,
  isActive = false,
  taskList,
  onOpenAddTask,
  onSubmit,
  onClick,
  onTaskDrop,
}: CardListProps) => {
  const { id = '', title, total, color, tasks } = taskList || {};

  const handleOpenAddTask = useCallback(() => {
    onOpenAddTask?.(id);
  }, []);

  const handleDragStart = useCallback(
    (event: DragEvent<HTMLDivElement>, taskId: string) => {
      event.dataTransfer.setData('taskId', taskId);
      event.dataTransfer.setData('sourceListId', id.toString());
    },
    [id],
  );

  const handleDragOver = (event: DragEvent<HTMLDivElement>) => {
    event.preventDefault();
  };

  const handleDrop = useCallback(
    (event: DragEvent<HTMLDivElement>) => {
      event.preventDefault();
      const taskId = event.dataTransfer.getData('taskId');

      onTaskDrop(taskId, id);
    },
    [id, onTaskDrop],
  );

  const memoizedTasks = useMemo(
    () =>
      tasks?.map((task) => {
        const {
          id,
          title,
          description,
          endDate,
          images,
          members,
          label,
          startDate,
        } = task;

        return (
          <CardItem
            key={task.id}
            id={id}
            title={title}
            description={description}
            label={label}
            startDate={startDate}
            endDate={endDate}
            images={images}
            members={members}
            onClick={onClick}
            onDragStart={handleDragStart}
          />
        );
      }),
    [tasks, onClick, handleDragStart],
  );

  return (
    <VStack
      gap={6}
      style={{ flexShrink: 0 }}
      onDragOver={handleDragOver}
      onDrop={handleDrop}
    >
      <HeaderList
        title={title}
        total={total}
        color={color}
        onClick={handleOpenAddTask}
        isDisabled={isActive}
      />

      <Flex
        gap={6}
        direction="column"
        h={825}
        overflowY="scroll"
        css={{ '&::-webkit-scrollbar': { display: 'none' } }}
      >
        {!!tasks?.length && memoizedTasks}

        {isActive && (
          <AddTaskForm
            isLoading={isLoading}
            onSubmit={onSubmit}
            onClose={handleOpenAddTask}
          />
        )}

        <Button
          variant="outline"
          display={isActive ? 'none' : 'flex'}
          justifyContent="flex-start"
          textTransform="uppercase"
          w={230}
          border="none"
          fontSize="base"
          onClick={handleOpenAddTask}
        >
          + new task
        </Button>
      </Flex>
    </VStack>
  );
};

export default memo(CardList);
