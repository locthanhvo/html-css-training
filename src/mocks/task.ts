import AvatarUser from '@/assets/images/avatar-small-1.png';
import AvatarUser2 from '@/assets/images/avatar-small-2.png';
import AvatarUser3 from '@/assets/images/avatar-small-3.png';

export const TASK_DETAIL = {
  id: '1',
  title: 'Global Resorts Network',
  label: [
    {
      name: 'Space Tasks 1',
      value: 'space-tasks-1',
    },
  ],
  members: [
    { value: '1', name: 'Member 1', image: AvatarUser },
    { value: '2', name: 'Member 2', image: AvatarUser2 },
    { value: '3', name: 'Member 3', image: AvatarUser3 },
  ],
  created_at: '2022-01-01',
  updated_at: '2022-01-01',
  startDate: '2024-07-01T00:00:00.000Z',
  endDate: '2024-07-03T00:00:00.000Z',
  boardId: '93d1f6a8-c312-436a-886f-a646bf424209',
};

export const TASK_DETAIL_2 = {
  id: '2',
  title: 'Global Suns Network',
  label: [
    {
      name: 'Space Tasks 1',
      value: 'space-tasks-1',
    },
    {
      name: 'Space Tasks 2',
      value: 'space-tasks-2',
    },
  ],
  members: [
    { value: '1', name: 'Member 1', image: AvatarUser },
    { value: '2', name: 'Member 2', image: AvatarUser2 },
    { value: '3', name: 'Member 3', image: AvatarUser3 },
  ],
  images: ['https://images.unsplash.com/photo-1555041469-a586c61ea9bc'],
  description: '<p>hello</p><p>hi</p>',
  startDate: '2024-07-01T00:00:00.000Z',
  endDate: '2024-07-02T00:00:00.000Z',
  created_at: '2022-01-01',
  updated_at: '2022-01-01',
  boardId: '93d1f6a8-c312-436a-886f-a646bf424209',
};

export const LIST_TASK = {
  id: '1',
  title: 'to do',
  total: 2,
  color: 'electricPurple',
  tasks: [TASK_DETAIL, TASK_DETAIL_2],
};

export const MOCK_TASKS = [
  {
    id: '550e8400-e29b-41d4-a716-446655440000',
    label: [
      {
        name: 'Space Tasks 1',
        value: 'space-tasks-1',
      },
    ],
    title: 'Global Suns Network',
    description: 'description 2 \n description 3',
    images: ['https://images.unsplash.com/photo-1555041469-a586c61ea9bc'],
    startDate: '2024-07-01T00:00:00.000Z',
    endDate: '2024-07-10T00:00:00.000Z',
    boardId: '93d1f6a8-c312-436a-886f-a646bf424209',
  },
  {
    id: '550e8400-e29b-41d4-a716-446655440001',
    label: [
      {
        name: 'Space Tasks 1',
        value: 'space-tasks-1',
      },
    ],
    title: 'Make Money Online Though',
    boardId: '93d1f6a8-c312-436a-886f-a646bf424209',
  },
  {
    id: '550e8400-e29b-41d4-a716-446655440002',
    label: [
      {
        name: 'Space Tasks 1',
        value: 'space-tasks-1',
      },
    ],
    title: 'Characteristics Of A Successful',
    description: 'Task 3 description',
    boardId: '93d1f6a8-c312-436a-886f-a646bf424209',
  },
];

export const MOCK_BOARDS = [
  {
    id: '93d1f6a8-c312-436a-886f-a646bf424209',
    title: 'Todo',
    color: 'electricPurple',
  },
  {
    id: '93d1f6a8-c312-436a-886f-a646bf424208',
    title: 'In Work',
    color: 'skyBlue',
  },
  {
    id: '93d1f6a8-c312-436a-886f-a646bf424207',
    title: 'Review',
    color: 'yellow',
  },
  {
    id: '9b1a1aa0-afc0-4bdb-90f7-7d8330ec254b',
    title: 'Done',
    color: 'green',
  },
];
