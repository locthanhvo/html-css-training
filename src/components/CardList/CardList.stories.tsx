import { Meta, StoryObj } from '@storybook/react';
import { BrowserRouter } from 'react-router-dom';

// Components
import CardList from '.';
import { TASK_DETAIL, TASK_DETAIL_2 } from '@/mocks';

export default {
  title: 'Components/CardList',
  component: CardList,
  decorators: [
    (Story) => (
      <BrowserRouter>
        <Story />
      </BrowserRouter>
    ),
  ],
  parameters: {
    layout: 'centered',
  },
} as Meta;

type Story = StoryObj<typeof CardList>;

export const Default: Story = {
  args: {
    taskList: {
      id: '1',
      title: 'to do',
      total: 2,
      color: 'electricPurple',
      tasks: [{ ...TASK_DETAIL }, { ...TASK_DETAIL_2 }],
    },
  },
};
