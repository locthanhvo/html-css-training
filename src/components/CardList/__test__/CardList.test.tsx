import { fireEvent, render, RenderResult } from '@testing-library/react';

// Components
import CardList from '..';

// Mocks
import { LIST_TASK } from '@/mocks';

const mockOnClick = jest.fn();

const mockOnSubmit = jest.fn();

const mockOnTaskDrop = jest.fn();

const mockSetData = jest.fn();

const mockGetData = jest.fn().mockImplementation((key) => {
  if (key === 'taskId') {
    return '2';
  }
  return '';
});

describe('CardList Component', () => {
  let renderResult: RenderResult;

  beforeEach(() => {
    renderResult = render(
      <CardList
        taskList={LIST_TASK}
        onSubmit={mockOnSubmit}
        onOpenAddTask={mockOnClick}
        onTaskDrop={mockOnTaskDrop}
      />,
    );
  });

  it('should match snapshot', () => {
    const { container } = renderResult;

    expect(container).toMatchSnapshot();
  });

  it('should call onClick add icon', async () => {
    const { getByRole } = renderResult;

    getByRole('button', { name: 'Add' }).click();

    expect(mockOnClick).toHaveBeenCalled();
  });

  it('should call onClick button add task', () => {
    const { getByRole } = renderResult;

    getByRole('button', { name: '+ new task' }).click();
    expect(mockOnClick).toHaveBeenCalled();
  });

  it('should call handleDragStart', () => {
    const { getByText } = renderResult;

    const dragEvent = {
      dataTransfer: {
        setData: mockSetData,
      },
    };

    fireEvent.dragStart(getByText('Global Suns Network'), dragEvent);
    expect(mockSetData).toHaveBeenCalledWith('taskId', '2');
    expect(mockSetData).toHaveBeenCalledWith('sourceListId', '1');
  });

  it('should call handleDrop', () => {
    const { getByText } = renderResult;

    const dropEvent = {
      dataTransfer: {
        getData: mockGetData,
      },
      preventDefault: jest.fn(),
    };

    fireEvent.drop(getByText('Global Suns Network'), dropEvent);

    expect(mockGetData).toHaveBeenCalledWith('taskId');
    expect(mockOnTaskDrop).toHaveBeenCalledWith('2', '1');
  });

  it('render AddTaskForm component', () => {
    const { getByPlaceholderText } = render(
      <CardList
        isActive={true}
        taskList={LIST_TASK}
        onSubmit={mockOnSubmit}
        onOpenAddTask={mockOnClick}
        onTaskDrop={mockOnTaskDrop}
      />,
    );

    expect(getByPlaceholderText('Task name or type')).toBeInTheDocument();
  });

  it('should call handleDragOver', () => {
    const mockPreventDefault = jest.fn();
    const { getByText } = renderResult;

    const dragOverEvent = {
      preventDefault: mockPreventDefault,
    };

    fireEvent.dragOver(getByText('Global Suns Network'), dragOverEvent);

    expect(getByText('Global Suns Network')).toBeInTheDocument();
  });

  it('should render task list empty', () => {
    const { container } = render(
      <CardList
        isActive={true}
        onSubmit={mockOnSubmit}
        onOpenAddTask={mockOnClick}
        onTaskDrop={mockOnTaskDrop}
      />,
    );

    expect(container).toBeInTheDocument();
  });
});
