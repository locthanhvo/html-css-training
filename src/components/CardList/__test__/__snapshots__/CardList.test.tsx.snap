// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList Component should match snapshot 1`] = `
<div>
  <div
    class="chakra-stack css-ylms7g"
    style="flex-shrink: 0;"
  >
    <div
      class="chakra-stack css-iti2te"
    >
      <div
        class="css-ayngmp"
      >
        <p
          class="chakra-text css-1kfhky2"
        >
          to do
        </p>
        <div
          class="css-1eqtwf1"
        >
          2
        </div>
      </div>
      <div
        class="css-k008qs"
      >
        <button
          class="chakra-button css-4xx2wk"
          type="button"
        >
          <svg
            fill="none"
            height="5"
            viewBox="0 0 16 5"
            width="16"
          >
            <path
              clip-rule="evenodd"
              d="M2.39999 0.633331C1.37333 0.633331 0.533325 1.47333 0.533325 2.5C0.533325 3.52666 1.37333 4.36666 2.39999 4.36666C3.42666 4.36666 4.26666 3.52666 4.26666 2.5C4.26666 1.47333 3.42666 0.633331 2.39999 0.633331ZM13.6 0.633331C12.5733 0.633331 11.7333 1.47333 11.7333 2.5C11.7333 3.52666 12.5733 4.36666 13.6 4.36666C14.6267 4.36666 15.4667 3.52666 15.4667 2.5C15.4667 1.47333 14.6267 0.633331 13.6 0.633331ZM7.99999 0.633331C6.97333 0.633331 6.13333 1.47333 6.13333 2.5C6.13333 3.52666 6.97333 4.36666 7.99999 4.36666C9.02666 4.36666 9.86666 3.52666 9.86666 2.5C9.86666 1.47333 9.02666 0.633331 7.99999 0.633331Z"
              fill="#C3CAD9"
              fill-rule="evenodd"
            />
          </svg>
        </button>
        <button
          class="chakra-button css-4xx2wk"
          type="button"
        >
          <svg
            fill="none"
            height="31"
            viewBox="0 0 30 31"
            width="30"
          >
            <path
              clip-rule="evenodd"
              d="M21 16.5H16V21.5C16 22.05 15.55 22.5 15 22.5C14.45 22.5 14 22.05 14 21.5V16.5H9C8.45 16.5 8 16.05 8 15.5C8 14.95 8.45 14.5 9 14.5H14V9.5C14 8.95 14.45 8.5 15 8.5C15.55 8.5 16 8.95 16 9.5V14.5H21C21.55 14.5 22 14.95 22 15.5C22 16.05 21.55 16.5 21 16.5Z"
              fill="#C3CAD9"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="css-b3z86p"
    >
      <div
        class="chakra-card css-12c38m6"
        data-testid="cart-item"
        draggable="true"
      >
        <div
          class="chakra-card__body css-old1by"
        >
          <img
            alt="Global Resorts Network image"
            class="chakra-image css-1pv0iot"
          />
          <div
            class="chakra-stack css-1kgz9m7"
          >
            <div
              class="css-4rk6vg"
            >
              <div
                class="css-nn803e"
              >
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-1hflwzw"
                  >
                    Space Tasks 1
                  </p>
                </div>
              </div>
              <div
                class="css-k008qs"
              >
                <div
                  class="css-k008qs"
                >
                  <img
                    alt="Avatar"
                    class="chakra-image css-1xtmofj"
                    src="test-file-stub"
                  />
                  <img
                    alt="Avatar"
                    class="chakra-image css-1xtmofj"
                    src="test-file-stub"
                  />
                  <img
                    alt="Avatar"
                    class="chakra-image css-1xtmofj"
                    src="test-file-stub"
                  />
                </div>
              </div>
            </div>
            <p
              class="chakra-text css-1p9c8n6"
            >
              Global Resorts Network
            </p>
            <div
              class="css-69i1ev"
            >
              <div
                class="css-lcy17d"
              >
                <svg
                  fill="none"
                  height="31"
                  viewBox="0 0 30 31"
                  width="30"
                >
                  <path
                    clip-rule="evenodd"
                    d="M7 12.5C7.55 12.5 8 12.05 8 11.5C8 10.95 7.55 10.5 7 10.5C6.45 10.5 6 10.95 6 11.5C6 12.05 6.45 12.5 7 12.5ZM8 15.5C8 16.05 7.55 16.5 7 16.5C6.45 16.5 6 16.05 6 15.5C6 14.95 6.45 14.5 7 14.5C7.55 14.5 8 14.95 8 15.5ZM8 19.5C8 20.05 7.55 20.5 7 20.5C6.45 20.5 6 20.05 6 19.5C6 18.95 6.45 18.5 7 18.5C7.55 18.5 8 18.95 8 19.5ZM24 15.5C24 16.05 23.55 16.5 23 16.5H11C10.45 16.5 10 16.05 10 15.5C10 14.95 10.45 14.5 11 14.5H23C23.55 14.5 24 14.95 24 15.5ZM23 20.5C23.55 20.5 24 20.05 24 19.5C24 18.95 23.55 18.5 23 18.5H11C10.45 18.5 10 18.95 10 19.5C10 20.05 10.45 20.5 11 20.5H23ZM11 12.5C10.45 12.5 10 12.05 10 11.5C10 10.95 10.45 10.5 11 10.5H23C23.55 10.5 24 10.95 24 11.5C24 12.05 23.55 12.5 23 12.5H11Z"
                    fill="#C3CAD9"
                    fill-rule="evenodd"
                  />
                </svg>
                <p
                  class="chakra-text css-8lo94t"
                />
              </div>
              <div
                class="css-lcy17d"
              >
                <svg
                  fill="none"
                  height="31"
                  viewBox="0 0 30 31"
                  width="30"
                >
                  <path
                    clip-rule="evenodd"
                    d="M20.2539 20H9.67393C7.58393 20 5.72393 18.47 5.52393 16.39C5.29393 14.01 7.16393 12 9.50393 12H21.8639C23.1739 12 24.3639 12.94 24.4939 14.24C24.6439 15.74 23.4739 17 22.0039 17H11.5039C10.9539 17 10.5039 16.55 10.5039 16C10.5039 15.45 10.9539 15 11.5039 15H20.2539C20.6639 15 21.0039 14.66 21.0039 14.25C21.0039 13.84 20.6639 13.5 20.2539 13.5H11.6439C10.3339 13.5 9.14393 14.44 9.01393 15.74C8.86393 17.24 10.0339 18.5 11.5039 18.5H21.8339C23.9239 18.5 25.7839 16.97 25.9839 14.89C26.2139 12.5 24.3439 10.5 22.0039 10.5H9.73393C6.86393 10.5 4.29393 12.6 4.02393 15.46C3.72393 18.75 6.28393 21.5 9.50393 21.5H20.2539C20.6639 21.5 21.0039 21.16 21.0039 20.75C21.0039 20.34 20.6639 20 20.2539 20Z"
                    fill="#C3CAD9"
                    fill-rule="evenodd"
                  />
                </svg>
                <p
                  class="chakra-text css-1hflwzw"
                />
              </div>
              <div
                class="css-69i1ev"
              >
                <svg
                  fill="none"
                  height="31"
                  viewBox="0 0 30 31"
                  width="30"
                >
                  <path
                    clip-rule="evenodd"
                    d="M18.4064 18.5314L15.0364 16.5314V12.0014C15.0364 11.6014 14.7164 11.2814 14.3164 11.2814H14.2564C13.8564 11.2814 13.5364 11.6014 13.5364 12.0014V16.7214C13.5364 17.0714 13.7164 17.4014 14.0264 17.5814L17.6764 19.7714C18.0164 19.9714 18.4564 19.8714 18.6564 19.5314C18.8664 19.1814 18.7564 18.7314 18.4064 18.5314ZM23.7164 8.29139L20.6364 5.73139C20.2164 5.38139 19.5864 5.43139 19.2264 5.86139C18.8764 6.28139 18.9364 6.91139 19.3564 7.27139L22.4264 9.83139C22.8464 10.1814 23.4764 10.1314 23.8364 9.70139C24.1964 9.28139 24.1364 8.65139 23.7164 8.29139ZM6.63638 9.83139L9.70638 7.27139C10.1364 6.91139 10.1964 6.28139 9.83638 5.86139C9.48638 5.43139 8.85638 5.38139 8.43638 5.73139L5.35638 8.29139C4.93638 8.65139 4.87638 9.28139 5.23638 9.70139C5.58638 10.1314 6.21638 10.1814 6.63638 9.83139ZM14.5364 7.28139C9.56638 7.28139 5.53638 11.3114 5.53638 16.2814C5.53638 21.2514 9.56638 25.2814 14.5364 25.2814C19.5064 25.2814 23.5364 21.2514 23.5364 16.2814C23.5364 11.3114 19.5064 7.28139 14.5364 7.28139ZM14.5364 23.2814C10.6764 23.2814 7.53638 20.1414 7.53638 16.2814C7.53638 12.4214 10.6764 9.28139 14.5364 9.28139C18.3964 9.28139 21.5364 12.4214 21.5364 16.2814C21.5364 20.1414 18.3964 23.2814 14.5364 23.2814Z"
                    fill="#C3CAD9"
                    fill-rule="evenodd"
                  />
                </svg>
                <p
                  class="chakra-text css-1hflwzw"
                >
                  2
                   Days Left
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="chakra-card css-12c38m6"
        data-testid="cart-item"
        draggable="true"
      >
        <div
          class="chakra-card__body css-old1by"
        >
          <img
            alt="Global Suns Network image"
            class="chakra-image css-1tg8l4p"
            src="https://images.unsplash.com/photo-1555041469-a586c61ea9bc"
          />
          <div
            class="chakra-stack css-1kgz9m7"
          >
            <div
              class="css-1ii69pb"
            >
              <div
                class="css-nn803e"
              >
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-1hflwzw"
                  >
                    Space Tasks 1
                  </p>
                </div>
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-1hflwzw"
                  >
                    Space Tasks 2
                  </p>
                </div>
              </div>
              <div
                class="css-k008qs"
              >
                <div
                  class="css-k008qs"
                >
                  <img
                    alt="Avatar"
                    class="chakra-image css-1xtmofj"
                    src="test-file-stub"
                  />
                  <img
                    alt="Avatar"
                    class="chakra-image css-1xtmofj"
                    src="test-file-stub"
                  />
                  <img
                    alt="Avatar"
                    class="chakra-image css-1xtmofj"
                    src="test-file-stub"
                  />
                </div>
              </div>
            </div>
            <p
              class="chakra-text css-1p9c8n6"
            >
              Global Suns Network
            </p>
            <div
              class="css-69i1ev"
            >
              <div
                class="css-69i1ev"
              >
                <svg
                  fill="none"
                  height="31"
                  viewBox="0 0 30 31"
                  width="30"
                >
                  <path
                    clip-rule="evenodd"
                    d="M7 12.5C7.55 12.5 8 12.05 8 11.5C8 10.95 7.55 10.5 7 10.5C6.45 10.5 6 10.95 6 11.5C6 12.05 6.45 12.5 7 12.5ZM8 15.5C8 16.05 7.55 16.5 7 16.5C6.45 16.5 6 16.05 6 15.5C6 14.95 6.45 14.5 7 14.5C7.55 14.5 8 14.95 8 15.5ZM8 19.5C8 20.05 7.55 20.5 7 20.5C6.45 20.5 6 20.05 6 19.5C6 18.95 6.45 18.5 7 18.5C7.55 18.5 8 18.95 8 19.5ZM24 15.5C24 16.05 23.55 16.5 23 16.5H11C10.45 16.5 10 16.05 10 15.5C10 14.95 10.45 14.5 11 14.5H23C23.55 14.5 24 14.95 24 15.5ZM23 20.5C23.55 20.5 24 20.05 24 19.5C24 18.95 23.55 18.5 23 18.5H11C10.45 18.5 10 18.95 10 19.5C10 20.05 10.45 20.5 11 20.5H23ZM11 12.5C10.45 12.5 10 12.05 10 11.5C10 10.95 10.45 10.5 11 10.5H23C23.55 10.5 24 10.95 24 11.5C24 12.05 23.55 12.5 23 12.5H11Z"
                    fill="#C3CAD9"
                    fill-rule="evenodd"
                  />
                </svg>
                <p
                  class="chakra-text css-8lo94t"
                >
                  2
                </p>
              </div>
              <div
                class="css-69i1ev"
              >
                <svg
                  fill="none"
                  height="31"
                  viewBox="0 0 30 31"
                  width="30"
                >
                  <path
                    clip-rule="evenodd"
                    d="M20.2539 20H9.67393C7.58393 20 5.72393 18.47 5.52393 16.39C5.29393 14.01 7.16393 12 9.50393 12H21.8639C23.1739 12 24.3639 12.94 24.4939 14.24C24.6439 15.74 23.4739 17 22.0039 17H11.5039C10.9539 17 10.5039 16.55 10.5039 16C10.5039 15.45 10.9539 15 11.5039 15H20.2539C20.6639 15 21.0039 14.66 21.0039 14.25C21.0039 13.84 20.6639 13.5 20.2539 13.5H11.6439C10.3339 13.5 9.14393 14.44 9.01393 15.74C8.86393 17.24 10.0339 18.5 11.5039 18.5H21.8339C23.9239 18.5 25.7839 16.97 25.9839 14.89C26.2139 12.5 24.3439 10.5 22.0039 10.5H9.73393C6.86393 10.5 4.29393 12.6 4.02393 15.46C3.72393 18.75 6.28393 21.5 9.50393 21.5H20.2539C20.6639 21.5 21.0039 21.16 21.0039 20.75C21.0039 20.34 20.6639 20 20.2539 20Z"
                    fill="#C3CAD9"
                    fill-rule="evenodd"
                  />
                </svg>
                <p
                  class="chakra-text css-1hflwzw"
                >
                  1
                </p>
              </div>
              <div
                class="css-69i1ev"
              >
                <svg
                  fill="none"
                  height="31"
                  viewBox="0 0 30 31"
                  width="30"
                >
                  <path
                    clip-rule="evenodd"
                    d="M18.4064 18.5314L15.0364 16.5314V12.0014C15.0364 11.6014 14.7164 11.2814 14.3164 11.2814H14.2564C13.8564 11.2814 13.5364 11.6014 13.5364 12.0014V16.7214C13.5364 17.0714 13.7164 17.4014 14.0264 17.5814L17.6764 19.7714C18.0164 19.9714 18.4564 19.8714 18.6564 19.5314C18.8664 19.1814 18.7564 18.7314 18.4064 18.5314ZM23.7164 8.29139L20.6364 5.73139C20.2164 5.38139 19.5864 5.43139 19.2264 5.86139C18.8764 6.28139 18.9364 6.91139 19.3564 7.27139L22.4264 9.83139C22.8464 10.1814 23.4764 10.1314 23.8364 9.70139C24.1964 9.28139 24.1364 8.65139 23.7164 8.29139ZM6.63638 9.83139L9.70638 7.27139C10.1364 6.91139 10.1964 6.28139 9.83638 5.86139C9.48638 5.43139 8.85638 5.38139 8.43638 5.73139L5.35638 8.29139C4.93638 8.65139 4.87638 9.28139 5.23638 9.70139C5.58638 10.1314 6.21638 10.1814 6.63638 9.83139ZM14.5364 7.28139C9.56638 7.28139 5.53638 11.3114 5.53638 16.2814C5.53638 21.2514 9.56638 25.2814 14.5364 25.2814C19.5064 25.2814 23.5364 21.2514 23.5364 16.2814C23.5364 11.3114 19.5064 7.28139 14.5364 7.28139ZM14.5364 23.2814C10.6764 23.2814 7.53638 20.1414 7.53638 16.2814C7.53638 12.4214 10.6764 9.28139 14.5364 9.28139C18.3964 9.28139 21.5364 12.4214 21.5364 16.2814C21.5364 20.1414 18.3964 23.2814 14.5364 23.2814Z"
                    fill="red"
                    fill-rule="evenodd"
                  />
                </svg>
                <p
                  class="chakra-text css-tokvmb"
                >
                  1
                   Days Left
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      class="chakra-button css-7j6h07"
      type="button"
    >
      + new task
    </button>
  </div>
</div>
`;
